src/controller/python/ChipDeviceController-ScriptPairingDeviceDiscoveryDelegate.cpp

```diff
@@ -32,6 +32,10 @@ void ScriptPairingDeviceDiscoveryDelegate::OnDiscoveredDevice(const Dnssd::Disco
     nodeData.resolutionData.ipAddress[0].ToString(buf);
     ChipLogProgress(chipTool, "Discovered Device: %s:%u", buf, port);
 
+    // Cancel discovery timer.
+    if (mActiveDeviceCommissioner->GetSystemLayer() != nullptr)
```

Comment:
Is there not a better place to get the system layer than adding a getter to CHIPDeviceController?

Reply:
Now I am using `chip::DeviceLayer::SystemLayer()`